src/main/java/com/epam/page/object/generator/PageObjectsGenerator.java

```diff
@@ -1,60 +1,153 @@
-    }
+    public void generatePageObjects(String jsonPath, String outputDir, List<String> urls)
+        throws IOException {
+
+        logger.info("Create list of RawSearchRules...");
+        List<RawSearchRule> rawSearchRuleList = rawSearchRuleMapper.getRawSearchRuleList(jsonPath);
+        logger.info("Finish creating list of RawSearchRules\n");
+
+        logger.info("Start validation RawSearchRules with using Json Schema...");
+        jsonSchemaValidator.validate(rawSearchRuleList);
+        logger.info("Finish validation!\n");
+
+        checker.checkRawSearchRules(rawSearchRuleList);
+
+        logger.info("Start transforming of RawSearchRules...\n");
+        List<SearchRule> searchRuleList = typeTransformer
+            .transform(rawSearchRuleList, selectorUtils, searchRuleExtractor);
+        logger.info("Finish transforming of RawSearchRules\n");
+
+        logger.info("Start Json validation...\n");
+        jsonValidators.validate(searchRuleList);
+        logger.info("Finish Json validation\n");
+
+        checker.checkSearchRules(searchRuleList);
+
+        List<WebPage> webPages = webPagesBuilder.generate(urls, searchRuleExtractor);
+        webPages.forEach(wp -> wp.addSearchRules(searchRuleList));
+        webValidators.validate(webPages);
 
-    private void generateJavaFiles(List<SearchRule> searchRules)
-        throws IOException, URISyntaxException, XpathToCssTransformerException {
+        List<JavaClassBuildable> rawJavaClasses = getJavaClassBuildables(webPages);
 
-        if (validatorsStarter.getValidationContext().hasInvalidRules()) {
+        List<IJavaClass> javaClasses = getJavaClasses(rawJavaClasses);
+
+        if (webPages.stream().anyMatch(WebPage::hasInvalidWebElementGroup)) {
             if (forceGenerateFile) {
-                javaFileWriter.writeFile(packageName, outPutDir, searchRules, urls);
+                javaFileWriter.writeFiles(outputDir, javaClasses);
+            }
+            StringBuilder stringBuilder = new StringBuilder("\n");
+            for (WebPage webPage : webPages) {
+                if (webPage.hasInvalidWebElementGroup()) {
+                    for (WebElementGroup webElementGroup : webPage.getWebElementGroups()) {
+                        webElementGroup.getValidationResults().stream()
+                            .filter(ValidationResult::isInvalid)
+                            .forEach(validationResultNew -> stringBuilder
+                                .append(validationResultNew.getReason()).append("\n"));
+                    }
+                }
             }
 
-            throw new ValidationException(validatorsStarter.getValidationContext());
+            throw new ValidationException(stringBuilder.toString());
         }
 
-        javaFileWriter.writeFile(packageName, outPutDir, searchRules, urls);
+        javaFileWriter.writeFiles(outputDir, javaClasses);
+    }
+
+    private List<IJavaClass> getJavaClasses(List<JavaClassBuildable> rawJavaClasses) {
```

Comment:
Could you change the name of the interface? not to use "I" as a prefix ?

Reply:
Done